### PR TITLE
Directions (Routes) and AR Polylines

### PR DESCRIPTION
**Pull request by intere**
Attempt at fixing merge conflicts with https://github.com/ProjectDent/ARKit-CoreLocation/pull/26.

The original update info is:
- a lot of code refactored
- added ability to display routes on AR view
- added a PR template
- added a `.travis.yml` to lint the pod
- added a 'changelog.md' file to track library updates / changes

### TODO
- [x] Factor `LocationNode` and `AnnotationNode` out of `LocationAnnotationNode.swift` and back into `LocationNode.swift`
- [x] Verify pod installation in an external project (I'll use my ARCL branch in GeoTrackKit for this)
- [x] Test the routing (I'll use my ARCL branch in GeoTrackKit for this)
- [x] Add a UI to test polyline routing in the example app

### Updates
- updates the swiftlint rules 
    - addresses many swiftlint issues in code
- makes many things more "swifty" (`forEach` function blocks instead of for loops for example)
- updated the `ARCL.podspec` file to contain the `swift_version`
- removed `.swift_version` file (Podspec is now responsible for it)
- adds the ability to display a route in the AR view

### Checklist

- [x] Appropriate label has been added to this PR (i.e., Bug, Enhancement, etc.).
- [x] Documentation has been added to all `open`, and `public` scoped methods and properties.
- [x] Changelog has been updated
- [x] Deferred ~Tests have have been added to all new features. (not a requirement, but helpful)~
- [x] Image/GIFs have been added for all UI related changed.

### Screenshots
First Demo: the polyline route demo:
route_polyline_demo

Second Demo: the (original) annootations
point_annotationos

### CODE REVIEW

File: ARCL/Source/SceneLocationViewEstimateDelegate.swift [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+@available(iOS 11.0, *)
+public protocol SceneLocationViewEstimateDelegate: class {
+    func didAddSceneLocationEstimate(sceneLocationView: SceneLocationView, position: SCNVector3, location: CLLocation)
+    func didRemoveSceneLocationEstimate(sceneLocationView: SceneLocationView, position: SCNVector3, location: CLLocation)
+}
+
+@available(iOS 11.0, *)
+public extension SceneLocationViewEstimateDelegate {
+    func didAddSceneLocationEstimate(sceneLocationView: SceneLocationView, position: SCNVector3, location: CLLocation) {
+        //
```
Comment (aaronbrethorst) [inline ARCL/Source/SceneLocationViewEstimateDelegate.swift]: Does this need to exist?

File: ARCL/Source/SceneLocationViewEstimateDelegate.swift [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+    func didSetupSceneNode(sceneLocationView: SceneLocationView, sceneNode: SCNNode)
+
+    func didUpdateLocationAndScaleOfLocationNode(sceneLocationView: SceneLocationView, locationNode: LocationNode)
+}
+
+@available(iOS 11.0, *)
+public extension SceneLocationViewDelegate {
+    func didAddSceneLocationEstimate(sceneLocationView: SceneLocationView, position: SCNVector3, location: CLLocation) {
+        //
```
Comment (aaronbrethorst) [inline ARCL/Source/SceneLocationViewEstimateDelegate.swift]: does this need to exist?

File: ARKit+CoreLocation/ViewController.swift [hunk 3]
```diff
@@ -101,114 +84,93 @@ class ViewController: UIViewController {
...
-                    self.locationEstimateAnnotation!.coordinate = bestLocationEstimate!.location.coordinate
-                } else {
-                    if self.locationEstimateAnnotation != nil {
-                        self.mapView.removeAnnotation(self.locationEstimateAnnotation!)
-                        self.locationEstimateAnnotation = nil
-                    }
-                }
-            }
-        }
+//        guard let currentLocation = sceneLocationView.currentLocation else { return }
```
Comment (aaronbrethorst) [inline ARKit+CoreLocation/ViewController.swift]: delete me! :)

File: ARKit+CoreLocation/ViewController.swift [hunk 4]
```diff
@@ -101,114 +84,93 @@ class ViewController: UIViewController {
...
-        if let heading = sceneLocationView.locationManager.heading,
-            let accuracy = sceneLocationView.locationManager.headingAccuracy {
-            infoLabel.text!.append("Heading: \(heading)º, accuracy: \(Int(round(accuracy)))º\n")
+        if let eulerAngles = sceneLocationView.currentEulerAngles {
+            infoLabel.text!.append("Euler x: \(eulerAngles.x.short), y: \(eulerAngles.y.short), z: \(eulerAngles.z.short)\n")
         }
 
-        let date = Date()
-        let comp = Calendar.current.dateComponents([.hour, .minute, .second, .nanosecond], from: date)
+//        if let heading = sceneLocationView.locationManager.heading,
```
Comment (aaronbrethorst) [inline ARKit+CoreLocation/ViewController.swift]: delete me :)

File: ARCL/Source/Location Manager/SceneLocationManager.swift [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+    func removeOldLocationEstimates() {
+        guard let currentScenePosition = sceneLocationDelegate?.scenePosition else { return }
+        removeOldLocationEstimates(currentScenePosition: currentScenePosition)
+    }
+
+    func removeOldLocationEstimates(currentScenePosition: SCNVector3) {
+        let currentPoint = CGPoint.pointWithVector(vector: currentScenePosition)
+
+        sceneLocationEstimates = sceneLocationEstimates.filter {
+            if #available(iOS 11.0, *) {
```
[the remark this answers is not in the document]
Comment (intere) [reply ARCL/Source/Location Manager/SceneLocationManager.swift]: I was thinking that too, however if we do that, won't that force all consumers of ARCL to require iOS 11?  I've created an issue for this: https://github.com/ProjectDent/ARKit-CoreLocation/issues/160

File: ARCL/Source/SceneLocationViewEstimateDelegate.swift (hunk 1 above)
Comment (intere) [reply ARCL/Source/SceneLocationViewEstimateDelegate.swift]: The idea is that they are default implementations of the functions (essentially making them optional when conforming to the protocol).  They don't have to exist, but by existing, they allow you to choose which functions you want to implement and not have no-op implementations for the ones you don't want to implement.

File: ARCL/Source/SceneLocationViewEstimateDelegate.swift (hunk 2 above)
Comment (intere) [reply ARCL/Source/SceneLocationViewEstimateDelegate.swift]: The idea is that they are default implementations of the functions (essentially making them optional when conforming to the protocol). They don't have to exist, but by existing, they allow you to choose which functions you want to implement and not have no-op implementations for the ones you don't want to implement.

**Comment by intere** [comment]
@aaronbrethorst - thanks for the feedback.  I've responded and/or updated based on your feedback.  I have started working on updating the UI and have hacked together the ability to specify a location, search for it, get directions and see those directions as a polyline in AR, however the settings UI is janky and I need to clean it up and polish it more.

**Pull request merged**